test/test_luaunit.lua

```diff
@@ -515,35 +513,54 @@ TestLuaUnitUtilities = { __class__ = 'TestLuaUnitUtilities' }
             { pattern={'toto', 'tutu'}, verbosity=lu.VERBOSITY_VERBOSE, output='tintin', testNames={'titi', 'tata', 'prout'}, fname='toto.xml' } )
 
-        lu.assertErrorMsgContains( 'option: -x', lu.LuaUnit.parseCmdLine, { '-x', } )
+        lu.assertErrorMsgContains( 'option: -$', lu.LuaUnit.parseCmdLine, { '-$', } )
     end
 
-    function TestLuaUnitUtilities:test_includePattern()
-        lu.assertEquals( lu.LuaUnit.patternInclude( nil, 'toto'), true )
-        lu.assertEquals( lu.LuaUnit.patternInclude( {}, 'toto'), false  )
-        lu.assertEquals( lu.LuaUnit.patternInclude( {'toto'}, 'toto'), true )
-        lu.assertEquals( lu.LuaUnit.patternInclude( {'toto'}, 'yyytotoxxx'), true )
-        lu.assertEquals( lu.LuaUnit.patternInclude( {'titi', 'toto'}, 'yyytotoxxx'), true )
-        lu.assertEquals( lu.LuaUnit.patternInclude( {'titi', 'to..'}, 'yyytoxxx'), true )
+    function TestLuaUnitUtilities:test_patternFilter()
+        lu.assertEquals( lu.private.patternFilter( nil, 'toto', true), true )
+        lu.assertEquals( lu.private.patternFilter( {}, 'toto', true), false  )
+        lu.assertEquals( lu.private.patternFilter( nil, 'titi', false), false )
+        lu.assertEquals( lu.private.patternFilter( {'toto'}, 'toto'), true )
+        lu.assertEquals( lu.private.patternFilter( {'toto'}, 'yyytotoxxx'), true )
+        lu.assertEquals( lu.private.patternFilter( {'titi', 'toto'}, 'yyytotoxxx'), true )
+        lu.assertEquals( lu.private.patternFilter( {'titi', 'to..'}, 'yyytoxxx'), true )
     end
 
     function TestLuaUnitUtilities:test_applyPatternFilter()
         local myTestToto1Value = { 'MyTestToto1.test1', MyTestToto1 }
 
-        local included, excluded = lu.LuaUnit.applyPatternFilter( nil, { myTestToto1Value } )
+        local included, excluded = lu.LuaUnit.applyPatternFilter( nil, nil, { myTestToto1Value } )
         lu.assertEquals( excluded, {} )
         lu.assertEquals( included, { myTestToto1Value } )
 
-        included, excluded = lu.LuaUnit.applyPatternFilter( {'T.to'}, { myTestToto1Value } )
+        included, excluded = lu.LuaUnit.applyPatternFilter( {'T.to'}, nil, { myTestToto1Value } )
         lu.assertEquals( excluded, {} )
         lu.assertEquals( included, { myTestToto1Value } )
 
-        included, excluded = lu.LuaUnit.applyPatternFilter( {'T.ti'}, { myTestToto1Value } )
+        included, excluded = lu.LuaUnit.applyPatternFilter( {'T.ti'}, nil, { myTestToto1Value } )
+        lu.assertEquals( excluded, { myTestToto1Value } )
+        lu.assertEquals( included, {} )
+
+        included, excluded = lu.LuaUnit.applyPatternFilter( nil, {'T.to'}, { myTestToto1Value } )
+        lu.assertEquals( excluded, { myTestToto1Value } )
+        lu.assertEquals( included, {} )
+
+        included, excluded = lu.LuaUnit.applyPatternFilter( {'T.t.'}, {'.o.o'}, { myTestToto1Value } )
         lu.assertEquals( excluded, { myTestToto1Value } )
         lu.assertEquals( included, {} )
     end
```

Comment:
Can you also add tests with multiple list of patterns, for both include / exclude ? And with probably more test references to filter against.